### ISSUE TITLE
[NeuralNetworkReductor] randomly shuffle training snapshots before splitting into training and validation set

### ISSUE THREAD

**Issue by HenKlei** [HenKlei/pymor#24]
In the `NeuralNetworkReductor`, the training data is split into training and validation set if no `validation_set` is provided. Before this is done, the training data should be shuffled randomly to obtain better training results. (Especially since the training set is often chosen uniformly on a grid which means that without random shuffling, the training set does not contain parameters from all over the domain).

**Comment by HenKlei** [comment]
Done in pymor/pymor/pull/1175.
